src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeUtils.cs

```diff
@@ -512,7 +512,9 @@ public static MethodInfo GetUserDefinedCoercionMethod(Type convertFrom, Type con
             if (retryForLifted)
             {
                 return FindConversionOperator(eMethods, nnExprType, nnConvType, implicitOnly)
-                    ?? FindConversionOperator(cMethods, nnExprType, nnConvType, implicitOnly);
+                    ?? FindConversionOperator(cMethods, nnExprType, nnConvType, implicitOnly)
+                    ?? FindConversionOperator(eMethods, nnExprType, convertToType, implicitOnly)
+                    ?? FindConversionOperator(eMethods, nnExprType, convertToType, implicitOnly);
             }
```

Comment:
The last  FindConversionOperator is the same as the one before. Did you mean cMethods ?
Also should have tests for that scenario.